trin-core/src/portalnet/overlay.rs

```diff
@@ -662,7 +662,11 @@ where
     }
 
     /// Performs a content lookup for `target`.
-    pub async fn lookup_content(&self, target: TContentKey) -> Option<Vec<u8>> {
+    /// Returns the target content along with the peers traversed during content lookup.
+    pub async fn lookup_content(
+        &self,
+        target: TContentKey,
+    ) -> (Option<Vec<u8>>, Option<Vec<NodeId>>) {
```

Comment:
Always returning a list of contacted peers (empty or not) makes sense to me. It does seem like useful information to return a `Result` for `lookup_content()`, but I'd rather implement that in a follow up pr to avoid bloat if you 👍